jupyter/nmm-game.ipynb

```diff
@@ -79,7 +102,41 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def nextStatesPhaseOne(s, p):\n",
+    "s0 = ((9, 9), (\n",
+    "    (' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '),\n",
+    "    (' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '),\n",
+    "    (' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ')\n",
+    "))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Folgezustände\n",
+    "Für die Definition des Spieles Mühle $G_{Nine Men's Morris}$ wird die Funktion `nextStates` benötigt. Diese nimmt einen Zustand `s` und einen Spieler `p` entgegen und errechnet mit diesen alle Folgezustände, die entstehen, wenn der gegebene Spieler einen legalen Zug spielt.\n",
+    "\n",
+    "Da das Spiel in drei Phasen aufgeteilt ist, ist auch die Implementierung aus Gründen der Übersicht in die folgenden Phasen unterteilt:\n",
+    "1. Die erste Phase des Spiels heißt _placing_ Phase, in der die Spieler alle ihre Steine aus dem Stapel auf das Spielbrett legen müssen. Hierbei dürfen bereits Mühlen gelegt und geschlagen werden. Es muss ein Stein plaziert werden.\n",
+    "2. Die zweite Phase heißt _moving_ Phase und beginnt sobald der Stapel des Spielers leer ist. Beide Spieler kommen gleichzeitig in die zweite Phase in der die eigenen Steine nur noch entlang der Linien auf die nächste Spielposition geschoben werden dürfen. Es muss ein Stein bewegt werden.\n",
+    "3. Die letzte Phase heißt _flying_ Phase. Diese Phase beginnt für einen Spieler, sobald dieser nur noch $3$ Steine auf dem Spielbrett liegen hat und sein Stapel leer ist. Der Eintritt in die dritte Phase ist nicht vom gegnerischen Spieler abhängig, somit können Spieler für mehrere Züge alleine in der dritten Phase sein. Hierbei können die verbleibenden Steine nun von jeder Position aufgehoben werden und auf jede freie Position bewegt werden, ohne dass diese direkt nebeneinander liegen müssen. Es muss ein Stein bewegt werden.\n",
+    "\n",
+    "In der ersten Phase, der _playing_ Phase, wird ein Stein vom Stapel genommen und auf ein freies Feld gelegt. Diese Züge werden in der Funktion `nextStatesPlace` implementiert, hierbei wird ein Spieler `p` und ein Zustand `s` erwartet, für angenommen wird, dass sich der aktuelle Zustand in der ersten Phase befindet.\n",
```

Comment:
ich glaube man spielt in jeder Phase :D

Reply:
Was willst du mir damit sagen? :D

Reply:
Ohhh. Got it :D Habe es gerade nochmal wegen einem anderen Kommentar lesen müssen :D